actors/builtin/miner/miner_actor.go

```diff
@@ -1498,6 +1498,13 @@ func (a Actor) ReportConsensusFault(rt Runtime, params *ReportConsensusFaultPara
 	burnAmount := big.Zero()
 	rewardAmount := big.Zero()
 	rt.State().Transaction(&st, func() {
+		info := getMinerInfo(rt, &st)
+
+		// verify miner hasn't already been faulted
+		if rt.CurrEpoch() < info.ConsensusFaultElapsed {
+			rt.Abortf(exitcode.ErrForbidden, "consensus fault has already been reported")
```

Comment:
Blocking: it's the `fault.Epoch` that must be after the exclusion period, in order to prevent replay.

Also the message isn't quite right - it might be a different report. Say something about expired or too old instead.